modules/models/record.php

```diff
@@ -62,29 +63,34 @@ protected static function loadRecordFromData($data)
 
     public static function getAll()
     {
-        return get_called_class()::where(array(1=>1));
+        return get_called_class()::where(array());
     }
 
     /*
         Usage: RecordName::where(array("name"=>"James", "student_id"=>12345));
     */
-    public static function where($attrs)
+    public static function where(array $attrs)
     {
         $table_name = get_called_class()::$table_name;
-        $sql_wheres = array();
-        foreach ($attrs as $key => $value) {
-            array_push($sql_wheres, sprintf("%s = '%s'", $key, $value));
-        }
 
-        $sql = sprintf(
-            "SELECT * FROM %s WHERE %s;",
-            $table_name,
-            implode(" AND ", $sql_wheres)
-        );
+        if (!empty($attrs)) {
+            $sql_wheres = array();
+            foreach ($attrs as $key => $value) {
+                array_push($sql_wheres, "$key = :$key");
+            }
+            $sql = sprintf(
+                "SELECT * FROM %s WHERE %s",
+                $table_name,
+                implode(" AND ", $sql_wheres)
+            );
+        } else {
+            $sql = sprintf(
+                "SELECT * FROM %s;",
+                $table_name
+            );
```

Comment:
I hate the fact that I had to repeat this, if you guys have any ideas im all ears